src/db/postgres/lrsql/postgres/record.clj

```diff
@@ -59,7 +60,13 @@
       (migrate-xapi-statement-timestamps! tx))
     (when-not (some? (query-xapi-statement-stored-exists tx))
       (alter-xapi-statement-add-stored! tx)
-      (migrate-xapi-statement-stored-times! tx)))
+      (migrate-xapi-statement-stored-times! tx))
+    (when-not (some? (query-state-document-last-modified-is-timestamptz tx))
+      (let [tz-input
+            {:tz-id (str "'" (.getID (.getTimeZone (Calendar/getInstance))) "'")}]
```

Comment:
This should definitely be extracted out into its own function. At the very least the ugly Java interop can go into the main `util` namespace. You could also in principle put the `:tz-id` wrapping in a separate `input/` namespace, though that would probably be overkill.

Reply:
actually i like that idea. keep the muck in input

Reply:
oooo wait a minute though this is engine-specific

Reply:
https://github.com/yetanalytics/lrsql/pull/300/commits/6dd5400dbd77ecaa5bd66617c1e9fce1f85fa653